packages/devtools_app/lib/src/performance/performance_screen.dart

```diff
@@ -375,6 +380,206 @@ class _SecondaryControls extends StatelessWidget {
   }
 }
 
+class EnhanceTracingButton extends StatefulWidget {
+  const EnhanceTracingButton({Key key}) : super(key: key);
+
+  @override
+  State<EnhanceTracingButton> createState() => _EnhanceTracingButtonState();
+}
+
+class _EnhanceTracingButtonState extends State<EnhanceTracingButton>
+    with AutoDisposeMixin {
+  static const _hoverYOffset = 10.0;
+
+  final _tracingServiceExtensions = [
+    profileWidgetBuilds,
+    profileRenderObjectLayouts,
+    profileRenderObjectPaints,
+  ];
+
+  final _tracingEnhanced = ValueNotifier(false);
+
+  List<bool> _extensionStates;
+
+  OverlayEntry _enhanceTracingOverlay;
+
+  bool _enhanceTracingOverlayHovered = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _extensionStates =
+        List.generate(_tracingServiceExtensions.length, (index) => false);
+    for (int i = 0; i < _tracingServiceExtensions.length; i++) {
+      final extension = _tracingServiceExtensions[i];
+      final state = serviceManager.serviceExtensionManager
+          .getServiceExtensionState(extension.extension);
+      _extensionStates[i] = state.value.enabled;
+      // Listen for extension state changes so that we can update the value of
+      // [_tracingEnhanced].
+      addAutoDisposeListener(state, () {
+        _extensionStates[i] = state.value.enabled;
+        _tracingEnhanced.value = _isTracingEnhanced();
+      });
+    }
+    _tracingEnhanced.value = _isTracingEnhanced();
+  }
+
+  bool _isTracingEnhanced() {
+    for (final state in _extensionStates) {
+      if (state) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder(
+      valueListenable: _tracingEnhanced,
+      builder: (context, tracingEnhanced, _) {
+        return DevToolsToggleButtonGroup(
+          children: const [
+            Padding(
+              padding: EdgeInsets.symmetric(horizontal: defaultSpacing),
+              child: MaterialIconLabel(
+                label: 'Enhance Tracing',
+                iconData: Icons.auto_awesome,
+                minScreenWidthForTextBeforeScaling:
+                    _SecondaryControls.minScreenWidthForTextBeforeScaling,
+              ),
+            ),
+          ],
+          selectedStates: [tracingEnhanced],
+          onPressed: (_) => _insertOverlay(context),
+        );
+      },
+    );
+  }
+
+  void _insertOverlay(BuildContext context) {
```

Comment:
maybe comment on the effect you are trying to achieve by adding this overlay like this.